codegenerator/cli/templates/static/codegen/src/globalState/GlobalState.res

```diff
@@ -918,8 +950,49 @@ let injectedTaskReducer = (
       }
     }
 
-    let throttler = state.writeThrottlers.pruneStaleData->ChainMap.get(chain)
-    throttler->Throttler.schedule(runPruneFunctions)
+    let throttler = state.writeThrottlers.pruneStaleEndBlockData->ChainMap.get(chain)
+    throttler->Throttler.schedule(runPrune)
+  | PruneStaleEntityHistory =>
+    let runPrune = async () => {
+      let safeChainIdAndBlockNumberArray =
+        state.chainManager->ChainManager.getSafeChainIdAndBlockNumberArray
+
+      if safeChainIdAndBlockNumberArray->Belt.Array.length > 0 {
+        let shouldDeepClean = if (
+          state.writeThrottlers.deepCleanCount ==
+            Env.ThrottleWrites.deepCleanEntityHistoryCycleCount
+        ) {
+          state.writeThrottlers.deepCleanCount = 0
+          true
+        } else {
+          state.writeThrottlers.deepCleanCount = state.writeThrottlers.deepCleanCount + 1
+          false
+        }
+        let timeRef = Hrtime.makeTimer()
+        await DbFunctions.sql->Postgres.beginSql(sql => {
+          Entities.allEntities->Belt.Array.map(entityMod => {
+            let module(Entity) = entityMod
+
+            sql->DbFunctions.EntityHistory.pruneStaleEntityHistory(
+              ~entityName=Entity.name,
+              ~safeChainIdAndBlockNumberArray,
+              ~shouldDeepClean,
+            )
+          })
+        })
+
+        if Env.Benchmark.shouldSaveData {
+          let elapsedTimeMillis = Hrtime.timeSince(timeRef)->Hrtime.toMillis->Hrtime.floatFromMillis
+
+          Benchmark.addSummaryData(
+            ~group="Other",
+            ~label="Prune Stale History Time (ms)",
+            ~value=elapsedTimeMillis,
+          )
+        }
+      }
+    }
+    state.writeThrottlers.pruneStaleEntityHistory->Throttler.schedule(runPrune)
```

Comment:
Am I right that it'll simply log if the transaction fails? For example, if we terminate it on our side.

Reply:
Yeah exactly it just logs